models/User.js

```diff
@@ -0,0 +1,62 @@
+import { Schema, model } from "mongoose";
+
+import { handleSaveError, setUpdateOptions } from "./hooks.js";
+import { emailRegExp } from "../constants/user-constants.js";
+
+const userSchema = new Schema(
+  {
+    userName: {
+      type: String,
+      unique: true,
+      default: null,
+    },
+    userEmail: {
+      type: String,
+      match: emailRegExp,
+      required: [true, "Email is required"],
+      unique: true,
+    },
+    userPassword: {
+      type: String,
+      required: [true, "Password is required"],
+    },
+    userAvatar: {
+      type: String,
+      default: null,
+    },
+    userGender: {
+      type: String,
+      default: null,
+    },
+    userWeight: {
+      type: Number,
+      default: null,
+    },
+    userActiveTime: {
+      type: Number,
+      default: null,
+    },
+    userWaterGoal: {
+      type: Number,
+      default: null,
+    },
+    trackerSetId: {
```

Comment:
this is expected to be: 
trackerSetId: {
      type: Schema.Types.ObjectId,
      ref: "water",
    },